src/interface/variable_pack.hpp

```diff
@@ -198,12 +204,17 @@ class PackIndexMap {
     shape_map_.insert(std::pair<std::string, vpack_types::Shape>(key, shape));
   }
 
-  vpack_types::FlatIdx GetFlatIdx(const std::string &key) {
+  vpack_types::FlatIdx GetFlatIdx(const std::string &key,
+                                  bool ThrowInvalidKeyError = true) {
```

Comment:
to stick with naming convention for variables, maybe do `throw_invalid_key_error`?